internal/rbd/encryption_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2023 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"github.com/ceph/ceph-csi/internal/util"
+	"testing"
+)
+
+func TestParseEncryptionOpts(t *testing.T) {
+	tests := []struct {
+		testName     string
+		volOptions   map[string]string
+		fallbackType util.EncryptionType
+		expectedKMS  string
+		expectedEnc  util.EncryptionType
+		expectedErr  error
+	}{
+		{
+			testName: "No Encryption Option",
+			volOptions: map[string]string{
+				"foo": "bar",
+			},
+			fallbackType: util.EncryptionTypeBlock,
+			expectedKMS:  "",
+			expectedEnc:  util.EncryptionTypeNone,
+			expectedErr:  nil,
+		},
+		{
+			testName: "Encrypted as false",
+			volOptions: map[string]string{
+				"encrypted": "false",
+			},
+			fallbackType: util.EncryptionTypeBlock,
+			expectedKMS:  "",
+			expectedEnc:  util.EncryptionTypeNone,
+			expectedErr:  nil,
+		},
+		{
+			testName: "Valid Encryption Option With KMS ID",
+			volOptions: map[string]string{
+				"encrypted":       "true",
```

Comment:
can you add an other test with `encrypted` set to an invalid string (not `true` or `false`)?

Reply:
Done, thanks!